src/signer.ts

```diff
@@ -471,20 +604,62 @@ export function createSafe(signer: Signer, provider = signer.provider, ownerAddr
       })
     },
 
+    /**
+     * Get the owner address of the current AvoSigner instance
+     * 
+     * @returns current AvoSigner instance owner's address
+     */
     async getOwnerAddress() {
       return await avoSigner.getOwnerAddress()
     },
 
+    /**
+     * Get the signer address of the current AvoSigner instance
+     * 
+     * @returns current AvoSigner instance signer's address
+     */
     async getSignerAddress() {
       return await avoSigner.getSignerAddress()
     },
 
+    /**
+     * Get the safe address of the current AvoSigner instance
+     * 
+     * @returns current avoSigner instance address
+     */
     async getSafeAddress() {
       return await avoSigner.getAddress()
     },
 
+    /**
+     * Get the current avoSafeNonce value at the smart wallet
+     * 
+     * @param chainId - The chain id of the network
+     * @returns current avoSafeNonce value
+     */
     async getSafeNonce(chainId: number | string) {
       return await avoSigner.getSafeNonce(Number(chainId))
-    }
+    },
+
+    /**
+     * Get the safe version of the current AvoSigner instance
+     * 
+     * @returns current avoSigner instance version
+     */
+    async getSafeVersion(chainId: number | string) {
+      return await avoContracts.safeVersion(chainId, await avoSigner.getAddress());
+    },
+
+    /**
+     * Get the status of a transaction receipt with success flag and error message
+     * 
+     * @param txReceipt the transaction receipt obtained e.g. by using `sendTransaction()` with returned `tx.wait()`
+     * @returns success flag and if false, also returns the error message as emitted by the `ExecuteFailed` event
+     */
+    async getTransactionReceiptStatus(txReceipt: TransactionReceipt): Promise<{ 
```

Comment:
@bergben I think we don't need this function. Instead, we can ask the avocado RPC `api_getTransactionByHash` method to get the details.

Reply:
Implemented: https://github.com/Instadapp/avocado-sdk/pull/6/commits/78704dd82ac49235898c7d62fff8b51bb5f546dd